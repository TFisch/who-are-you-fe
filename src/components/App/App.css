body {
  font-family: 'Josefin Slab', serif;
  font-variant: small-caps;
  background: url("../../images/reincarnation.jpg") no-repeat center;
  background-size: cover;
  height: 100vh;
  width: 100%;
  margin: 0; }

button:focus {
  outline: none;
  border: none; }

h2 {
  width: 12rem;
  text-align: center;
  text-overflow: ellipsis; }

h3 {
  font-size: 1.5rem;
  margin: 1rem auto; }

input {
  border-radius: 0.8rem;
  font-family: 'Josefin Slab', serif;
  height: 2rem;
  border: 0.1rem solid black;
  width: 10rem;
  padding-left: 1rem;
  font-size: 1rem; }

input:focus {
  outline: none; }

label {
  width: auto;
  margin: auto;
  width: 12rem;
  font-size: 1.5rem;
  font-family: 'Josefin Slab', serif;
  font-variant: small-caps; }
