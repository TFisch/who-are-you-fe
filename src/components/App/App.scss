body {
  font-family: 'Josefin Slab', serif;
  font-variant: small-caps;
}

button:focus {
  outline: none;
  border: none;
}
