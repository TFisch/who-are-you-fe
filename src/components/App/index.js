import React, { Component } from 'react';
import DateSubmitForm from '../common/DateSubmitForm';
import Header from '../Header';
import ReincarnationDisplay from '../ReincarnationDisplay';
import CardContainer from '../common/CardContainer';
import './App.css';
import {
  fetchDateId,
  fetchUsers,
  fetchDeathByDate,
  deleteUser
} from '../../utilities/apiCalls';

class App extends Component {
  constructor() {
    super();
    this.state = {
      username: '',
      showSubmitForm: true,
      deathsByDate: {},
      showReincarnatedUsers: false,
      showReincarnation: false,
      showReincarnationDisplay: false,
      cleanUsers: []
    };
  }

  findDeathMatch = async (cleanedDate, year) => {
    const dateId = await fetchDateId(cleanedDate, year);
    const deathsByDate = await fetchDeathByDate(dateId, year);
    await this.setState({ deathsByDate, showReincarnationDisplay: true });
  };

  hideForm = () => {
    this.setState({ showSubmitForm: false });
  };

  showForm = () => {
    this.setState({
      showSubmitForm: true,
      showReincarnation: false,
      showReincarnationDisplay: false
    });
  };

  handleDeathSubmit = (cleanedDate, year, username) => {
    this.hideForm();
    this.setState({ username });
    this.findDeathMatch(cleanedDate, year);
  };

  showReincarnatedUsers = () => {
    this.setState({
      showReincarnatedUsers: true,
      showReincarnationDisplay: false
    });
  };

  // showReincarnationDisplay = () => {
  //   this.setState({ showReincarnationDisplay: true });
  // };

  getCleanUsers = async () => {
    const cleanUsers = await fetchUsers();
    this.setState({ cleanUsers });
    console.log(this.state.showReincarnatedUsers);
  };

  handleClick = async e => {
    deleteUser(e.target.id);
    e.target.parentNode.remove();
  };

  render() {
    const {
      showSubmitForm,
      deathsByDate,
      username,
      showReincarnatedUsers,
      showReincarnationDisplay
    } = this.state;
    return (
      <div>
        <Header
          hideForm={this.hideForm}
          showForm={this.showForm}
          getCleanUsers={this.getCleanUsers}
          showReincarnatedUsers={this.showReincarnatedUsers}
        />

        {showSubmitForm && (
          <DateSubmitForm
            headerText="WHO ARE YOU?"
            inputOneText="What is your name?"
            inputTwoText="What is your birthday?"
            handleDeathSubmit={this.handleDeathSubmit}
            showReincarnation={this.showReincarnation}
          />
        )}

        {showReincarnationDisplay && (
          <ReincarnationDisplay
            deathsByDate={deathsByDate}
            username={username}
            tryAgain={this.showForm}
            getCleanUsers={this.getCleanUsers}
            showReincarnatedUsers={this.showReincarnatedUsers}
          />
        )}

        {showReincarnatedUsers && (
          <CardContainer
            users={this.state.cleanUsers}
            handleClick={this.handleClick}
            showReincarnatedUsers={this.showReincarnatedUsers}
          />
        )}
      </div>
    );
  }
}

export default App;
