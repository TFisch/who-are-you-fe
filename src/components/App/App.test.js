import React from 'react';
import { shallow } from 'enzyme';
import App from './';

describe('App', () => {
  let wrapper;

  beforeEach(() => {

    wrapper = shallow(<App />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});