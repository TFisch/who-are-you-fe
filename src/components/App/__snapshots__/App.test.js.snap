// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          getCleanUsers={[Function]}
          hideForm={[Function]}
          showForm={[Function]}
          showReincarnatedUsers={[Function]}
        />,
        <DateSubmitForm
          handleDeathSubmit={[Function]}
          headerText="WHO ARE YOU?"
          inputOneText="what is your name?"
          inputTwoText="what is your birthday?"
        />,
        false,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "getCleanUsers": [Function],
          "hideForm": [Function],
          "showForm": [Function],
          "showReincarnatedUsers": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleDeathSubmit": [Function],
          "headerText": "WHO ARE YOU?",
          "inputOneText": "what is your name?",
          "inputTwoText": "what is your birthday?",
          "showReincarnation": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      false,
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            getCleanUsers={[Function]}
            hideForm={[Function]}
            showForm={[Function]}
            showReincarnatedUsers={[Function]}
          />,
          <DateSubmitForm
            handleDeathSubmit={[Function]}
            headerText="WHO ARE YOU?"
            inputOneText="what is your name?"
            inputTwoText="what is your birthday?"
          />,
          false,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "getCleanUsers": [Function],
            "hideForm": [Function],
            "showForm": [Function],
            "showReincarnatedUsers": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleDeathSubmit": [Function],
            "headerText": "WHO ARE YOU?",
            "inputOneText": "what is your name?",
            "inputTwoText": "what is your birthday?",
            "showReincarnation": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        false,
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
